stablehlo/reference/Element.cpp

```diff
@@ -607,6 +601,203 @@ Element ceil(const Element &el) {
+    return Element(type, std::complex<APFloat>(realVal, imagVal));
+  }
+  report_fatal_error(invalidArgument("Unsupported element type: %s",
+                                     debugString(type).c_str()));
+}
+
+Element convert(Type type, uint64_t value) {
+  if (isSupportedBooleanType(type)) return Element(type, value != 0);
+  if (isSupportedIntegerType(type))
+    return Element(type,
+                   APInt(type.getIntOrFloatBitWidth(), value,
+                         /*isSigned=*/isSupportedSignedIntegerType(type)));
+  if (isSupportedFloatType(type)) {
+    bool roundingErr;
+    auto floatVal = APFloat(static_cast<double>(value));
+    floatVal.convert(type.cast<FloatType>().getFloatSemantics(),
+                     APFloat::rmNearestTiesToEven, &roundingErr);
+    return Element(type, floatVal);
+  }
+  if (isSupportedComplexType(type)) {
+    const llvm::fltSemantics &fltSemantics = type.cast<ComplexType>()
+                                                 .getElementType()
+                                                 .cast<FloatType>()
+                                                 .getFloatSemantics();
+    bool roundingErr;
+    auto realVal = APFloat(static_cast<double>(value));
+    realVal.convert(fltSemantics, APFloat::rmNearestTiesToEven, &roundingErr);
+    auto imagVal = APFloat(0.0);
+    imagVal.convert(fltSemantics, APFloat::rmNearestTiesToEven, &roundingErr);
+    return Element(type, std::complex<APFloat>(realVal, imagVal));
+  }
+  report_fatal_error(invalidArgument("Unsupported element type: %s",
+                                     debugString(type).c_str()));
+}
+
+Element convert(Type type, APFloat value) {
+  if (isSupportedBooleanType(type)) return Element(type, value.isNonZero());
+  if (isSupportedIntegerType(type))
+    return Element(
+        type, llvm::APIntOps::RoundDoubleToAPInt(value.convertToDouble(),
+                                                 type.getIntOrFloatBitWidth()));
+  if (isSupportedFloatType(type)) {
+    bool roundingErr;
+    value.convert(type.cast<FloatType>().getFloatSemantics(),
+                  APFloat::rmNearestTiesToEven, &roundingErr);
+    return Element(type, value);
+  }
+  if (isSupportedComplexType(type)) {
+    const llvm::fltSemantics &fltSemantics = type.cast<ComplexType>()
+                                                 .getElementType()
+                                                 .cast<FloatType>()
+                                                 .getFloatSemantics();
+    bool roundingErr;
+    value.convert(fltSemantics, APFloat::rmNearestTiesToEven, &roundingErr);
+    auto imagVal = APFloat(0.0);
+    imagVal.convert(fltSemantics, APFloat::rmNearestTiesToEven, &roundingErr);
+    return Element(type, std::complex<APFloat>(value, imagVal));
+  }
+  report_fatal_error(invalidArgument("Unsupported element type: %s",
+                                     debugString(type).c_str()));
+}
+
+Element convert(Type type, double value) {
```

Comment:
There's a lot of duplication between the APFloat and double overloads. Can we simplify this overload by creating an APFloat with the given value / the double semantics and then calling the APFloat overload?